### PR TITLE
Ensure MRTK core systems support enable/disable correctly

### PR DESCRIPTION
**Pull request by Troy-Ferrell**
## Overview

This change adds an isEnabled property at the BaseCoreSystem layer to allow the MRTK core systems to check if they have been enabled or disabled and respond appropriately.

Most of the larger code chunk changes are just moving Obsolete methods to the bottom of the class.

**NOTE** This PR targets a separate branch than MRTK_Development. This is to break up the "feature" into multiple PRs and not impact MRTK_dev

toggle-diagnostics

```C#
        if (isEnabled)
            CoreServices.DiagnosticsSystem.Disable();
        else
            CoreServices.DiagnosticsSystem.Enable();
```

## Changes
- Fixes: #5901 
Related: #7100

### CODE REVIEW

File: Assets/MixedRealityToolkit/Services/BaseCoreSystem.cs [hunk 1]
```diff
@@ -1,39 +1,42 @@
...
-            IMixedRealityServiceRegistrar registrar,
-            BaseMixedRealityProfile profile = null) : this(profile)
+            BaseMixedRealityProfile profile = null) : base()
         {
-            Registrar = registrar;
+            ConfigurationProfile = profile;
+            Priority = 5; // Core systems have a higher default priority than other services
         }
 
+        #region Obsolete
```
Comment (wiwei) [inline Assets/MixedRealityToolkit/Services/BaseCoreSystem.cs]: > #region Obsolete [](start = 7, length = 17)

Just to check, all that was done here was a reorder (i.e moving the Obsolete constructor second, adding it to an Obsolete region) right?

The diffs are actually fairly hard to read despite being what seems like just a reorder

File: Assets/MixedRealityToolkit/Services/BaseDataProvider.cs [hunk 2]
```diff
@@ -5,49 +5,74 @@
 
 namespace Microsoft.MixedReality.Toolkit
 {
-    [System.Obsolete("Add a <T> of type IMixedRealityService, which defines the service type this data provider is valid for.")]
-    public abstract class BaseDataProvider : BaseDataProvider<IMixedRealityService>
+    /// <summary>
+    /// The base data provider implements <see cref="IMixedRealityDataProvider"/> and provides default properties for all data providers.
+    /// </summary>
+    public abstract class BaseDataProvider<T> : BaseService, IMixedRealityDataProvider where T : IMixedRealityService
```
Comment (wiwei) [inline Assets/MixedRealityToolkit/Services/BaseDataProvider.cs]: >public [](start = 4, length = 6)

Same question around reorder - if it's just a reorder then this lgtm.

**Comment by wiwei** [comment]
>             base.LateUpdate();

I think that this section would also need an !IsEnabled guard as well - I forget exactly what services had some stuff here but I thought there was some non-trivial stuff here.

Anyways would be great to have it here too!

---
Refers to: Assets/MixedRealityToolkit/Services/BaseDataProviderAccessCoreSystem.cs:73 in 4913415. [](commit_id = 49134151540a1cf698743b0a1b6d379a6ac22bf5, deletion_comment = False)

File: Assets/MixedRealityToolkit/Services/BaseCoreSystem.cs (hunk 1 above)
Comment (Troy-Ferrell) [reply Assets/MixedRealityToolkit/Services/BaseCoreSystem.cs]: yes, sorry I tried to call that out in the description

Comment (wiwei) [reply Assets/MixedRealityToolkit/Services/BaseCoreSystem.cs]: gah, that's my bad for missing that then.

---
In reply to: [370892079](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/7145#discussion_r370892079) [](ancestors = 370892079)

**Comment by wiwei** [comment]
>             base.LateUpdate();

Similar comment about having this be in the !IsEnabled guard as well

---
Refers to: Assets/MixedRealityToolkit.Services/InputSimulation/InputSimulationService.cs:305 in 4913415. [](commit_id = 49134151540a1cf698743b0a1b6d379a6ac22bf5, deletion_comment = False)

**Comment by wiwei** [comment]
I can spin up a PR validation run on this manually on the backend when you want, just lemme know when you want it.

**Comment by rajandeepavontus** [comment]
Thanks, guys! This was really helpful!!!!

**Comment by Troy-Ferrell** [comment]
@wiwei can you run PR? thanks!

**Comment by wiwei** [comment]
fdsagdasg

**Comment by wiwei** [comment]
> fdsagdasg

This is remembering that /azp run doesn't watch this branch so I had to manually kick it off:

https://dev.azure.com/aipmr/MixedRealityToolkit-Unity-CI/_build/results?buildId=8877&view=results

**Comment by wiwei** [comment]
Ack, it looks like there's some issues with the pull request negotiation code for non-mainline /azp run ones - I can look to patch that up, but it might be easier to go in + then run mrtk_ci afterward.

**Comment by Troy-Ferrell** [comment]
@wiwei ack, I'll merge and then run CI.

**Pull request merged**